control-plane/pkg/reconciler/source/source.go

```diff
@@ -144,15 +147,40 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, ks *sources.Kafk
 			Extensions: ks.Spec.CloudEventOverrides.Extensions,
 		}
 	}
+
 	if kt, ok := ks.Labels[sources.KafkaKeyTypeLabel]; ok && len(kt) > 0 {
 		expectedCg.Spec.Template.Spec.Configs.KeyType = &kt
 	}
 
+	// TODO: make these parameters below configurable and maybe unexposed
+	if ks.Annotations != nil {
```

Comment:
Different question: why are we using annotations for this on the ConsumerGroup API? Can we use a proper spec structure on the ConsumerGroup API?

Reply:
like:

```golang
type ConsumerGroupLagAutoscaling struct {
  MinScale uint64
  // etc
}
```

Reply:
Sounds good to me, at least for now.

Perhaps the suggested improvement is something we can do (shortly) _after_ ?

Reply:
Yes - adding this to the list.